File: 2022/1_Calorie_Counting/part_1.rb
input =
  File
    .open('input.txt')
    .read
    .split("\n\n")
    .map { |bag| bag.split("\n").map { |line| line.to_i } }

bags_with_total_calories = input.map { |bag| bag.reduce(:+) }

print bags_with_total_calories.sort.reverse.first

File: 2022/1_Calorie_Counting/part_1.ts
import { readFileSync } from 'fs'

const input: number[][] = readFileSync('input.txt', 'utf-8')
  .split('\n\n')
  .map((elfPack) => elfPack.split('\n').map((candyBar) => parseInt(candyBar)))

// --------------

const bagsWithTotalCallories = input.map((bag) =>
  bag.reduce((sum, number) => sum + number)
)

const bagWithMostCallories = bagsWithTotalCallories.sort((a, b) => b - a)[0]

console.log(bagWithMostCallories)
